tests/perf/allocator_perf.cc

```diff
@@ -0,0 +1,139 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2023 ScyllaDB
+ */
+
+#include <seastar/testing/perf_tests.hh>
+
+#include <seastar/core/memory.hh>
+#include <sys/mman.h>
+
+struct allocator_test {
+    static constexpr size_t COUNT = 1000;
+    std::array<void *, COUNT> _pointers{};
+};
+
+struct allocator_test_split : public allocator_test {
+    enum alloc_test_flags {
+      MEASURE_ALLOC = 1 << 0,
+      MEASURE_FREE = 2 << 0,
+    };
+
+    template <int M> struct maybe_measure {
```

Comment:
bool? what's M?

Reply:
No please change the name M to something meaningful. I know templates are supposed to be difficult but they do allow multi-character names.

And you could narrow the int to bool by writing something != 0.

Reply:
After another look at this I was writing 20 lines to save 5 (numbers approximate), so I've refactored this a bit and the NTTP has disappeared and I think it's cleaner overall. Thanks for the eyes on this.